packages/ia-components/components/audio-players/archive-audio-jwplayer-wrapper.jsx

```diff
@@ -124,19 +132,19 @@ class ArchiveAudioPlayer extends Component {
     const isNowMainPlayer = !prevSourceData.hasOwnProperty('index') && trackN;
     const unsupportedAlbumView = incomingTrackNum === 0; // 0 = album for 3d party players
 
-    if (!jwplayerInstance
-      || unsupportedAlbumView
-      || Number.isNaN(trackN)
-      || (trackN < 0)
-      || isNowMainPlayer
-    ) {
+    if (!jwplayerInstance || unsupportedAlbumView) {
+      return;
+    }
+
+    if (Number.isNaN(trackN) || (trackN < 0) || isNowMainPlayer) {
       return;
     }
 
     const playerStatus = jwplayerInstance.getState();
+    const jwpTrackIndex = jwplayerInstance.getPlaylistIndex();
     const jwpPlayingOnColdLoad = prevIndex === null && incomingTrackNum >= 1;
     const incomingTrackChange = !jwpPlayingOnColdLoad && (incomingTrackNum > prevIndex) || (trackNumber !== incomingTrackNum);
-    const autoplaying = incomingTrackChange && (playerStatus === 'idle');
+    const autoplaying = incomingTrackChange && (playerStatus === 'idle') && (jwpTrackIndex + 1 === incomingTrackNum);
```

Comment:
checking directly with JWP for its current track `jwpTrackIndex + 1 === incomingTrackNum` is the needed check to support `autoplay` notion.

for the next refactor upwards, the main player class (Play8 or equiv) can have a set of events to emit so we no longer have to connect to JWP (or whatever is under the hood)